src/common/src/array/arrow.rs

```diff
@@ -152,44 +155,51 @@ impl From<arrow_schema::DataType> for DataType {
     }
 }
 
-impl From<&DataType> for arrow_schema::DataType {
-    fn from(value: &DataType) -> Self {
+impl TryFrom<&DataType> for arrow_schema::DataType {
+    type Error = String;
```

Comment:
Should this also be `ArrayError::ToArrow`?

Reply:
I think it's a low level operation. It will be call by other module and the error type will be define by the module call it. Such as:
https://github.com/risingwavelabs/risingwave/blob/f48cd6dbac2152718326ee801ecd3c6580468c83/src/frontend/src/handler/create_function.rs#L135C55-L135C55

Actually I'm confused whether it's appropriate to place it in array module.🤔